### PR TITLE
For #3424 - Make talkback ignore ImageButton in Collection Selection by removing contentDescription and setting importantForAccessibility=no

### LINKED ISSUE #3424: FNX3-17004 ⁃ [Bug][a11y] '+' button from Add new collection read as Close button using Talkback
## Prerequisites
You have Talkback option enabled from your Android settings.

## Steps to reproduce
1. Open Firefox preview;
1. Open a page;
1. Tap 3dots menu;
1. Tap on 'Save to Collection' button;
1. Make Talkback to read the '+' button from Add new collection.

### Expected behavior
Talkback read correctly the '+' button.

### Actual behavior
Talkback read the '+' button as Close button.

### Device information
* Android device: LG G7fit (Android 8.1).
* Fenix version: 1.0.1924.

### PR DESCRIPTION
**Pull request by JorgeDLS**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/collections/CollectionCreationUIView.kt [hunk 1]
```diff
@@ -208,9 +208,8 @@ class CollectionCreationUIView(
                 val drawable = view.context.getDrawable(R.drawable.ic_new)
                 drawable?.setTint(ContextCompat.getColor(view.context, R.color.photonWhite))
                 view.bottom_bar_icon_button.setImageDrawable(drawable)
-                view.bottom_bar_icon_button.setOnClickListener {
-                    actionEmitter.onNext(CollectionCreationAction.AddNewCollection)
-                }
+                view.bottom_bar_icon_button.contentDescription = null
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/collections/CollectionCreationUIView.kt]: This code change seems redundant with the XML change and actually performs a redundant UI change that costs a couple CPU cycles. Is there any reason we need both?

Comment (JorgeDLS) [reply app/src/main/java/org/mozilla/fenix/collections/CollectionCreationUIView.kt]: I'm not sure if it's needed; I saw that pattern too, same changes in the XML and programmatically for the same attributes (ImageButton's drawable, tint, text, etc) so I followed it.

Do you want me to remove one of them (XML or programmatically)?

Comment (JorgeDLS) [reply app/src/main/java/org/mozilla/fenix/collections/CollectionCreationUIView.kt]: @colintheshots I'd be glad to know your thoughts about this ⏫

Comment (colintheshots) [reply app/src/main/java/org/mozilla/fenix/collections/CollectionCreationUIView.kt]: Yes, please remove the programmatic change. The XML should apply when the View gets created.

**Comment by JorgeDLS** [comment]
Comments resolved, please take a look :)

**Comment by ekager** [comment]
So when the user taps on the "+" icon for add new collection, does the layout beind get the click to add a new collection? What about for "X"? Because this icon changes functionality depending on what screen we're on we want to make sure it works for all screens.

**Comment by JorgeDLS** [comment]
> So when the user taps on the "+" icon for add new collection, does the layout beind get the click to add a new collection? What about for "X"? Because this icon changes functionality depending on what screen we're on we want to make sure it works for all screens.

Yes, the layout behind the `ImageButton` gets the click properly. Also, this change doesn't modify the "X" state behavior, due to `component_collection_creation_select_collection` file is the only one changed.

**Comment by ekager** [comment]
Sorry for the delay! I'm still seeing this read as "Close Button" when testing with TalkBack on. Are you interested in continuing work on this?

Because of the way the layouts are set up, you can't just change the layout file `component_collection_creation_select_collection`, you'll need to change the content description of this button in the UIView programatically depending on the mode we're in.

**Comment by JorgeDLS** [comment]
@ekager you're right, my first commit did those changes, but (probably for my bad understanding) I was told to remove it as it was redundant.

I've done the changes again, sorry for the error.

**Comment by JorgeDLS** [comment]
> One more thing and then we're good! You'll also have to add the opposite dynamic content description and important for accessiblity in the SaveCollectionStep.SelectTabs area as well for when the user presses back from the SelectCollection step.
> 
> ```
> view.bottom_bar_icon_button.contentDescription = $closebuttonstring
> view.bottom_bar_icon_button.importantForAccessibility = yes
> ```

I've just done these changes, forgive me for the delay, I'm on vacation right now, so it has been challenging to find some moment to do it :)

Thanks for your help!

**Pull request merged**